### PR TITLE
Detect database name from property file

### PR DESCRIPTION
**Pull request by rujche**
Detect database name from property file.

### CODE REVIEW

File: cli/azd/internal/appdetect/spring_boot.go [hunk 1]
```diff
@@ -250,12 +251,58 @@ func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndPro
...
+		return
+	}
+	if azdProject.Metadata.DatabaseNameInPropertySpringDatasourceUrl == nil {
+		azdProject.Metadata.DatabaseNameInPropertySpringDatasourceUrl = map[DatabaseDep]string{}
+	}
+	if strings.HasPrefix(propertyValue, "jdbc:postgresql") {
+		azdProject.Metadata.DatabaseNameInPropertySpringDatasourceUrl[DbPostgres] = databaseName
+	} else if strings.HasPrefix(propertyValue, "jdbc:mysql") {
+		azdProject.Metadata.DatabaseNameInPropertySpringDatasourceUrl[DbMySql] = databaseName
+	}
```
Comment (rujche) [inline cli/azd/internal/appdetect/spring_boot.go:282]: I'll apply to all other databases after this PR merged.

File: cli/azd/internal/appdetect/spring_boot.go [hunk 2]
```diff
@@ -286,8 +333,8 @@ func detectDependencySpringCloudAzureStarterJdbcMysql(azdProject *Project, sprin
 }
 
 func detectPropertySpringApplicationName(azdProject *Project, springBootProject *SpringBootProject) {
-	var targetSpringAppName = "spring.application.name"
-	if appName, ok := springBootProject.applicationProperties[targetSpringAppName]; ok {
```
Comment (rujche) [inline cli/azd/internal/appdetect/spring_boot.go:290]: It's an error before this PR.

File: cli/azd/internal/appdetect/spring_boot_property.go [hunk 3]
```diff
@@ -114,11 +115,24 @@ func readPropertiesInPropertiesFile(propertiesFilePath string, result map[string
...
+			continue
+		}
+		envVar := match[1]
+		defaultValue := match[2]
+		value := os.Getenv(envVar)
+		if value == "" {
+			value = defaultValue
+		}
+		placeholder := match[0]
+		result = strings.Replace(result, placeholder, value, -1)
```
Comment (rujche) [inline cli/azd/internal/appdetect/spring_boot_property.go:135]: Please refer to the unit test to see why this change is necessary.

**Review by haoozhang — APPROVED** [review]

File: cli/azd/internal/appdetect/spring_boot_property_test.go [hunk 4]
```diff
@@ -56,6 +56,24 @@ func TestGetEnvironmentVariablePlaceholderHandledValue(t *testing.T) {
...
+			"defaultValue",
+		},
+		{
+			"Has valid environment variable placeholder with default value, and environment variable set",
+			"${VALUE_THREE:defaultValue}",
+			map[string]string{"VALUE_THREE": "valueThree"},
+			"valueThree",
+		},
+		{
+			"Has multiple environment variable placeholder with default value, and environment not variable set",
```
[the remark this answers is not in the document]
Comment (rujche) [reply cli/azd/internal/appdetect/spring_boot_property_test.go:72]: Thanks, will fix it in next PR.
